---
 defs.h    |   26 +++++++++++++-------------
 symbols.c |    2 +-
 2 files changed, 14 insertions(+), 14 deletions(-)

--- a/defs.h
+++ b/defs.h
@@ -3702,7 +3702,7 @@ tint text_value_cache_byte(tulong, unsig
 void dump_text_value_cache(tint);
 void clear_text_value_cache(void);
 void dump_numargs_cache(void);
-tint patch_kernel_symbol(struct gnu_request *);
+int patch_kernel_symbol(struct gnu_request *);
 struct syment *generic_machdep_value_to_symbol(tulong, tulong *);
 tlong OFFSET_verify(tlong, char *, char *, tint, char *);
 tlong SIZE_verify(tlong, char *, char *, tint, char *);
@@ -4957,26 +4957,26 @@ void display_gdb_banner(void);
 void get_gdb_version(void);
 void gdb_session_init(void);
 void gdb_interface(struct gnu_request *);
-tint gdb_pass_through(char *, FILE *, tulong);
-tint gdb_readmem_callback(tulong, void *, tint, tint);
+int gdb_pass_through(char *, FILE *, bfd_vma);
+int gdb_readmem_callback(bfd_vma, void *, int, int);
 void gdb_error_hook(void);
 void restore_gdb_sanity(void);
-tint is_gdb_command(tint, tulong);
-char *gdb_command_string(tint, char *, tint);
+int is_gdb_command(int, bfd_vma);
+char *gdb_command_string(int, char *, tint);
 void dump_gnu_request(struct gnu_request *, tint);
-tint gdb_CRASHDEBUG(tulong);
+int gdb_CRASHDEBUG(bfd_vma);
 void dump_gdb_data(void);
 #if defined(GDB_6_0) || defined(GDB_6_1) || defined(GDB_7_0)
 void update_gdb_hooks(void);
 #endif
 void gdb_readnow_warning(void);
-extern tint *gdb_output_format;
-extern tuint *gdb_print_max;
-extern tint *gdb_prettyprint_structs;
-extern tint *gdb_prettyprint_arrays;
-extern tint *gdb_repeat_count_threshold;
-extern tint *gdb_stop_print_at_null;
-extern tuint *gdb_output_radix;
+extern int *gdb_output_format;
+extern unsigned int *gdb_print_max;
+extern int *gdb_prettyprint_structs;
+extern int *gdb_prettyprint_arrays;
+extern int *gdb_repeat_count_threshold;
+extern int *gdb_stop_print_at_null;
+extern unsigned int *gdb_output_radix;
 
 /*
  *  gdb/top.c
--- a/symbols.c
+++ b/symbols.c
@@ -10140,7 +10140,7 @@ clear_text_value_cache(void)
 
 #define last_sp addr2
 
-tint 
+int 
 patch_kernel_symbol(struct gnu_request *req)
 {
 	tint i, c;
