---
 unwind_x86_64.h |    2 --
 vas_crash.h     |    3 ---
 2 files changed, 5 deletions(-)

--- a/unwind_x86_64.h
+++ b/unwind_x86_64.h
@@ -13,7 +13,6 @@
 #define CONFIG_64BIT 1
 #define NULL ((void *)0)
 
-typedef unsigned long size_t;
 typedef unsigned char u8;
 typedef signed short s16;
 typedef unsigned short u16;
--- a/vas_crash.h
+++ b/vas_crash.h
@@ -50,9 +50,6 @@ void save_core(void);
 typedef int boolean_t;
 #endif
 
-#define TRUE 1
-#define FALSE 0
-
 
 
 /* mem structure */
