---
 defs.h |   36 ++++++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

--- a/defs.h
+++ b/defs.h
@@ -107,6 +107,42 @@
 #define NR_CPUS  (4)
 #endif
 
+#if defined(_32BIT_)
+
+typedef  int16_t tshort __attribute__ ((aligned (2)));
+typedef uint16_t tushort __attribute__ ((aligned (2)));
+typedef  int32_t tint __attribute__ ((aligned (4)));
+typedef uint32_t tuint __attribute__ ((aligned (4)));
+typedef  int32_t tlong __attribute__ ((aligned (4)));
+typedef uint32_t tulong __attribute__ ((aligned (4)));
+typedef  int64_t tlonglong __attribute__ ((aligned (8)));
+typedef uint64_t tulonglong __attribute__ ((aligned (8)));
+typedef uint32_t tptr __attribute__ ((aligned (4)));
+
+#elif defined(_64BIT_)
+
+typedef  int16_t tshort __attribute__ ((aligned (2)));
+typedef uint16_t tushort __attribute__ ((aligned (2)));
+typedef  int32_t tint __attribute__ ((aligned (4)));
+typedef uint32_t tuint __attribute__ ((aligned (4)));
+typedef  int64_t tlong __attribute__ ((aligned (8)));
+typedef uint64_t tulong __attribute__ ((aligned (8)));
+typedef  int64_t tlonglong __attribute__ ((aligned (8)));
+typedef uint64_t tulonglong __attribute__ ((aligned (8)));
+typedef uint64_t tptr __attribute__ ((aligned (8)));
+
+#endif
+
+#else	/* GDB_COMMON */
+
+/* Not quite correct, but seems to work */
+typedef bfd_vma tulong;
+typedef bfd_signed_vma tlong;
+
+#endif	/* GDB_COMMON */
+
+#ifndef GDB_COMMON
+
 #define BUFSIZE  (1500)
 #define NULLCHAR ('\0')
 
