---
 defs.h |   43 +++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 43 insertions(+)

--- a/defs.h
+++ b/defs.h
@@ -50,6 +50,7 @@
 #include <regex.h>
 #include <endian.h>
 #include "gdb-7.3.1/bfd/bfd.h" /* bfd_vma types */
+#include <inttypes.h>
 
 #ifndef ATTRIBUTE_UNUSED
 #define ATTRIBUTE_UNUSED __attribute__ ((__unused__))
@@ -125,6 +126,27 @@ typedef  int64_t tlonglong __attribute__
 typedef uint64_t tulonglong __attribute__ ((aligned (8)));
 typedef uint32_t tptr __attribute__ ((aligned (4)));
 
+/* tint and tuint format specifiers: */
+#define PRId	PRId32
+#define PRIi	PRIi32
+#define PRIo	PRIo32
+#define PRIu	PRIu32
+#define PRIx	PRIx32
+
+/* tlong and tulong format specifiers: */
+#define PRIld	PRId32
+#define PRIli	PRIi32
+#define PRIlo	PRIo32
+#define PRIlu	PRIu32
+#define PRIlx	PRIx32
+
+/* tlonglong and tulonglong specifiers: */
+#define PRIlld	PRId64
+#define PRIlli	PRIi64
+#define PRIllo	PRIo64
+#define PRIllu	PRIu64
+#define PRIllx	PRIx64
+
 #elif defined(_64BIT_)
 
 typedef  int16_t tshort __attribute__ ((aligned (2)));
@@ -137,6 +159,27 @@ typedef  int64_t tlonglong __attribute__
 typedef uint64_t tulonglong __attribute__ ((aligned (8)));
 typedef uint64_t tptr __attribute__ ((aligned (8)));
 
+/* tint and tuint format specifiers: */
+#define PRId	PRId32
+#define PRIi	PRIi32
+#define PRIo	PRIo32
+#define PRIu	PRIu32
+#define PRIx	PRIx32
+
+/* tlong and tulong format specifiers: */
+#define PRIld	PRId64
+#define PRIli	PRIi64
+#define PRIlo	PRIo64
+#define PRIlu	PRIu64
+#define PRIlx	PRIx64
+
+/* tlonglong and tulonglong specifiers: */
+#define PRIlld	PRId64
+#define PRIlli	PRIi64
+#define PRIllo	PRIo64
+#define PRIllu	PRIu64
+#define PRIllx	PRIx64
+
 #endif
 
 typedef tlong	toff_t;
