---
 unwind_x86_32_64.c |    4 ++--
 unwind_x86_64.h    |    2 --
 2 files changed, 2 insertions(+), 4 deletions(-)

--- a/unwind_x86_32_64.c
+++ b/unwind_x86_32_64.c
@@ -17,6 +17,8 @@
  * DWARF CFI written by Jan Beulich
  */
 
+#include "defs.h"
+
 #ifdef X86_64
 #include "unwind_x86_64.h"
 #endif
@@ -24,8 +26,6 @@
 #include "unwind_x86.h"
 #endif
 
-#include "defs.h"
-
 #define MAX_STACK_DEPTH 8
 
 static struct local_unwind_table {
--- a/unwind_x86_64.h
+++ b/unwind_x86_64.h
@@ -13,7 +13,6 @@
 #define CONFIG_64BIT 1
 #define NULL ((void *)0)
 
-typedef tulong size_t;
 typedef unsigned char u8;
 typedef signed short s16;
 typedef unsigned short u16;
@@ -58,7 +57,6 @@ extern tint unwind(struct unwind_frame_i
 extern void init_unwind_table(void);
 extern void free_unwind_table(void);
 
-#define offsetof(TYPE, MEMBER) ((size_t) &((TYPE *)0)->MEMBER)
 #define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]))
 #define BUILD_BUG_ON(condition) ((void)sizeof(char[1 - 2*!!(condition)]))
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { tint:-!!(e); }))
